community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/cardinality/assumeIndependence/PatternSelectivityCalculator.scala

```diff
@@ -87,7 +87,7 @@ case class PatternSelectivityCalculator(stats: GraphStatistics, combiner: Select
               }
           val selectivityPerLength = selectivityPerLengthAndStep.flatMap(combiner.andTogetherSelectivities)
           combiner.orTogetherSelectivities(selectivityPerLength).
-            getOrElse(throw new InternalException("Unexpectedly tried to calculate cardinality of a [*0..0] relationship"))
+            getOrElse(Selectivity.ZERO)
```

Comment:
The case for `*..0` and `*1..0` are equivalent (i.e. when no lower bound is provided, it is assigned to be the value of 1). This is not equivalent to `*0..0`. `(n)-*1..0` and `(n)-*0..0` have different behaviours; the former returns nothing and the latter returns `n` only.